hpe3parclient/client.py

```diff
@@ -4816,3 +4816,28 @@ def getScheduleStatus(self, schedule_name):
                 return 'active'
         msg = "Couldn't find the schedule '%s' status" % schedule_name
         raise exceptions.SSHException(reason=msg)
+
+    @staticmethod
+    def getPortNumber(ip, login, password, port=22,
+                      conn_timeout=None, privatekey=None,
+                      **kwargs):
+        """Set SSH Options for ssh calls.
+
+        This is used to set the SSH credentials for calls
+        that use SSH instead of REST HTTP.
+
+        :param 3PAR credentials
+        :return: 443: If build version starts with 4.x.x
+        :        8080: If build version not starts with 4.x
+        """
+        ssh_client = ssh.HPE3PARSSHClient(ip, login, password, port,
+                                        conn_timeout, privatekey,
+                                        **kwargs)
+        ssh_client.open()
+        cmd = ['showversion', '-b']
+        version=ssh_client.run(cmd)
+        port_number=8080
+        build_version = ''.join(version[0].split(" ")[2].split('-')[0].split('.'))
+        if build_version.startswith('4'):
+            port_number=443
+        return port_number
```

Comment:
Do `return 443 if build_version.startswith('4') else 8080` instead.
By this, you won't need line number 4839, 4841 and 4842.

Before returning please close the ssh connection.
```
if ssh_client :
            ssh_client .close()
```

Reply:
@ansarars please add unit tests.

Reply:
@ansarars - preferred way of opening/closing connection is to put it in try/finally block so that the connection is closed even if any exception occurs.